lib/browser/api/web-contents.js

```diff
@@ -379,23 +373,21 @@ WebContents.prototype._init = function () {
 
   // Dispatch IPC messages to the ipc module.
   this.on('-ipc-message', function (event, internal, channel, args) {
+    addReplyToEvent(event, internal)
     if (internal) {
-      addReplyInternalToEvent(event)
       ipcMainInternal.emit(channel, event, ...args)
     } else {
-      addReplyToEvent(event)
       this.emit('ipc-message', event, channel, ...args)
       ipcMain.emit(channel, event, ...args)
     }
   })
 
   this.on('-ipc-message-sync', function (event, internal, channel, args) {
     addReturnValueToEvent(event)
+    addReplyToEvent(event, internal)
```

Comment:
it looks like this adds an internal reply no matter what? is this an intentional change>

Reply:
it always adds `event.reply()`, which `_sendToFrameInternal` for internal messages or `sendToFrame` for public messages.